cpp/include/cudf/functions.h

```diff
@@ -796,6 +796,42 @@ gdf_error gdf_min(gdf_column *col, void *dev_result, gdf_size_type dev_result_si
  * --------------------------------------------------------------------------*/
 gdf_error gdf_max(gdf_column *col, void *dev_result, gdf_size_type dev_result_size);
 
+/* --------------------------------------------------------------------------*
+ * @brief  Computes the rolling window function of the values in a column.
+ *
+ * This matches Pandas' API for DataFrame.rolling with a couple notable
+ * differences:
+ * - instead of the center flag it uses the forward window size to allow for
+ * more flexible windows. The total window size = window + forward_window.
+ * Element i uses elements [i-window+1, i+forward_window] to do the window
+ * computation.
+ * - instead of storing NA/NaN for output rows that do not meet the minimum
+ * number of observations this function updates the valid bitmask of the column
+ * to indicate which elements are valid.
+ *
+ * @param[out] output_col The output column
+ * @param[in] input_col The input column
+ * @param[in] window The rolling window size, output_col[i] accumulates values
+ *                from input_col[i-window+1] to input_col[i] inclusive
+ * @param[in] min_periods Minimum number of observations in window required to
+ *                have a value, otherwise 0 is stored in the valid bit mask for
+ *                the element i
+ * @param[in] forward_window The window size in the forward direction,
+ *                output_col[i] accumulates values from input_col[i] to
+ *                input_col[i+forward_window] inclusive
+ * @param[in] agg_type The rolling window aggregtion type (sum, max, min, etc.)
+ *
+ * @return    GDF_SUCCESS if the operation was successful, otherwise an
+ *            appropriate error code.
+ *
+ * --------------------------------------------------------------------------*/
+gdf_error gdf_rolling_window(gdf_column *output_col,
+			     const gdf_column *input_col,
+			     gdf_size_type window,
+			     gdf_size_type min_periods,
+			     gdf_size_type forward_window,
+			     gdf_agg_op agg_type);
```

Comment:
Rolling Windows normally have other operators that are not your traditional aggregators here are some examples
Value

FIRST_VALUE()
LAG()
LAST_VALUE()
LEAD()
Aggregate

AVG()
COUNT()
MAX()
MIN()
SUM()
Ranking

CUME_DIST()
DENSE_RANK()
NTILE()
PERCENT_RANK()
RANK()
ROW_NUMBER()

Reply:
Ok that makes sense. It would be good to keep a list somewhere of what we might eventually want to include.